rpcserver.go

```diff
@@ -1820,6 +1826,51 @@ func GetUnconfirmedBalance(w *wallet.Wallet, chainSvr *chain.Client, icmd interf
 	return (unconfirmed - confirmed).ToBTC(), nil
 }
 
+// ImportAddress handles an importaddress request by parsing
+// a encoded address and importing it as a watch-only address to the
+// imported address account.
+func ImportAddress(w *wallet.Wallet, chainSvr *chain.Client, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*btcjson.ImportAddressCmd)
+
+	addr, err := decodeAddress(cmd.Address, activeNet.Params)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = w.ImportAddress(addr, nil, *cmd.Rescan)
+	return nil, err
+}
+
+// ImportPubKey handles an importpubkey request by parsing a public key and
+// adding it to the imported account.
+func ImportPubKey(w *wallet.Wallet, chainSvr *chain.Client, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*btcjson.ImportPubKeyCmd)
+
+	pubKeyBytes, err := decodeHexStr(cmd.PubKey)
+	if err != nil {
+		return nil, err
+	}
+
+	var compressed bool
+	switch n := len(pubKeyBytes); n {
```

Comment:
I think this whole switch can be removed.  The correct length is checked by `ParsePubKey` below.  All you would need to pass to `ImportPubKey` is the bool `len(pubKeyBytes) == btcec.PubKeyBytesLenCompressed`.

I kind of think that the btcec type should save whether the pubkey was compressed or not, but oh well.